packages/react-reconciler/src/__tests__/ReactOwnerStacks-test.js

```diff
@@ -85,4 +95,289 @@ describe('ReactOwnerStacks', () => {
+    });
+
+    // Our internal act flushes pending timers, so this will render with owner
+    // stacks intact until we hit the limit again.
+    await act(() => {
+      ReactNoop.render(
+        <App
+          // TODO: Owner Stacks should update on re-render.
+          key="two"
+          siblingsBeforeStackOne={499}
+        />,
+      );
+    });
+
+    expect({
+      pendingTimers: jest.getTimerCount(),
+      stackOne: normalizeCodeLocInfo(stackOne),
+      stackTwo: normalizeCodeLocInfo(stackTwo),
+    }).toEqual({
+      pendingTimers: 0,
+      // rendered everything before cutoff
+      stackOne: '\n    in App (at **)',
+      stackTwo: '\n    in App (at **)',
+    });
+  });
+
+  // @gate __DEV__
+  it('Fiber: resets the owner stack limit periodically', async () => {
+    function App({siblingsBeforeStackOne, timeout}) {
+      const children = [];
+      for (
+        let i = 0;
+        i <
+        siblingsBeforeStackOne -
+          // JSX callsites before this are irrelevant since we always reset the limit
+          // when we start a render. Both <App /> and <UnknownOwner /> were created
+          // before we actually start rendering.
+          0 -
+          // Stop so that OwnerStackOne will be right before cutoff
+          1;
+        i++
+      ) {
+        children.push(<Component key={i} />);
+      }
+      children.push(<OwnerStackOne key="stackOne" />);
+      children.push(<OwnerStackDelayed key="stackTwo" timeout={timeout} />);
+
+      return children;
+    }
+
+    function Component() {
+      return null;
+    }
+
+    let stackOne;
+    function OwnerStackOne() {
+      Scheduler.log('render OwnerStackOne');
+      stackOne = React.captureOwnerStack();
+    }
+
+    let stackTwo;
+    function OwnerStackTwo() {
+      Scheduler.log('render OwnerStackTwo');
+      stackTwo = React.captureOwnerStack();
+    }
+    function OwnerStackDelayed({timeout}) {
+      Scheduler.log('render OwnerStackDelayed');
+      React.use(timeout);
+      return <OwnerStackTwo />;
+    }
+
+    React.startTransition(() => {
+      ReactNoop.render(
+        <App
+          key="one"
+          // Should be the value with of `ownerStackLimit` with `__VARIANT__` so that we see the cutoff
+          siblingsBeforeStackOne={500}
+          timeout={
+            new Promise(resolve =>
+              setTimeout(
+                resolve,
+                // Must be greater or equal then the reset interval
+                1000,
+              ),
+            )
+          }
+        />,
+      );
+    });
+
+    await waitFor(['render OwnerStackOne', 'render OwnerStackDelayed']);
+
+    expect({
+      pendingTimers: jest.getTimerCount(),
+      stackOne: normalizeCodeLocInfo(stackOne),
+      stackTwo: normalizeCodeLocInfo(stackTwo),
+    }).toEqual({
+      // 1 for the timeout
+      // And since we haven't committed yet, we continue to reset the Owner Stack
+      // limit periodically.
+      pendingTimers: 2,
```

Comment:
This is where Fiber differs from Fizz and Flight. `prepareFreshStack` isn't called when we render from a ping if we never committed. Fizz and Flight will stop at this point in the test though.